### DISCUSSION TITLE
Code can be optimized?

### DISCUSSION THREAD

**Discussion by andrew-kuzovov** [dotnet/orleans#6980] [General]
I was looking into the Orleans code and it seems to me that this condition is always true:
`if (workItemGroup?.TaskScheduler is TaskScheduler scheduler)` - [link](https://github.com/dotnet/orleans/blob/a1eebad4c4ecca9de24d25d697d7413686842b32/src/Orleans.Runtime/Scheduler/OrleansTaskScheduler.cs#L134)
`ActivationTaskScheduler TaskScheduler ` property is actually not nullable and always initialized to ActivationTaskScheduler instance, right?
the same redundant checks also can be found in other places

**Comment by ReubenBond** [comment]
Thanks for the input, Andrew. The value might be null, but the code could be changed to `var scheduler = workItemGroup?.TaskScheduler;` with a conditional guard of `scheduler is not null` to avoid performing the type check. The `workItemGroup?.TaskScheduler is {} scheduler` syntax shortens that further. Would you be open to submitting a PR?

**Reply by andrew-kuzovov** [reply]
Thanks for reply Reuben.
I can try to submit a PR, but I still don't understand how it's possible that the value can become null (except maybe some reflection magic), could you clarify a bit?)

**Reply by ReubenBond** [reply]
GetWorkItemGroup can return null

**Reply by ReubenBond** [reply]
We should enable start enabling nullability checks.

**Reply by andrew-kuzovov** [reply]
ahh ok, thx

**Reply by andrew-kuzovov** [reply]
@ReubenBond do I need some permissions to submit a PR?

**Reply by ReubenBond** [reply]
No, no permission is needed to open a PR. If it's a big PR, then it is often a good idea to submit a draft PR or a discussion issue first so that we can try to decide if we can accept it before too much effort is put into it.
